pkg/controller/registry/resolver/resolver.go

```diff
@@ -23,18 +24,20 @@ type Resolver interface {
 }
 
 type OperatorsV1alpha1Resolver struct {
-	subLister v1alpha1listers.SubscriptionLister
-	csvLister v1alpha1listers.ClusterServiceVersionLister
-	client    versioned.Interface
+	subLister  v1alpha1listers.SubscriptionLister
+	csvLister  v1alpha1listers.ClusterServiceVersionLister
+	client     versioned.Interface
+	kubeclient kubernetes.Interface
 }
 
 var _ Resolver = &OperatorsV1alpha1Resolver{}
 
-func NewOperatorsV1alpha1Resolver(lister operatorlister.OperatorLister, client versioned.Interface) *OperatorsV1alpha1Resolver {
+func NewOperatorsV1alpha1Resolver(lister operatorlister.OperatorLister, client versioned.Interface, kubeclient kubernetes.Interface) *OperatorsV1alpha1Resolver {
```

Comment:
If we think the parameter list will keep growing, then it's probably worth it to use the variadic config options pattern here.

On the flip side, we could also figure out how to compose typed clientsets nicely (see https://github.com/kubernetes/client-go/blob/0bdba2f9188006fc64057c2f6d82a0f9ee0ee422/kubernetes/scheme/register.go#L121).